### PR TITLE
set validationwebhookconfig as Fail close by default for remote clusters

### PR DESCRIPTION
**Pull request by irisdingbj**
when we use istiod-less as default remote for multi-cluster, we have below error:

```
info	validationController	Not ready to switch validation to fail-closed: dummy invalid rejected for the wrong reason: gateways.networking.istio.io is forbidden: User "system:serviceaccount:istio-system:istio-reader-service-account" cannot create resource "gateways" in API group "networking.istio.io" in the namespace "istio-system"
```

### CODE REVIEW

File: manifests/charts/base/templates/role.yaml [hunk 1]
```diff
@@ -18,3 +18,20 @@ rules:
...
+  name: istio-reader-{{ .Values.global.istioNamespace }}
+  labels:
+    app: istio-reader
+    release: {{ .Release.Name }}
+rules:
+  # permissions to verify the webhook is ready and rejecting
+  # invalid config. We use --server-dry-run so no config is persisted.
+  - apiGroups: ["networking.istio.io"]
+    verbs: ["create"]
+    resources: ["gateways"]
```
Comment (linsun) [inline manifests/charts/base/templates/role.yaml:36]: ideally we would have a separate non reader role for this, perhaps in a follow up PR?

Comment (linsun) [reply manifests/charts/base/templates/role.yaml:36]: Also, wonder if this is really needed if we terminate the connection at the gw and have real certs at the gw.

**Comment by irisdingbj** [comment]
@howardjohn @linsun PTAL, Thanks.

File: pkg/webhooks/validation/controller/controller.go [hunk 2]
```diff
@@ -364,6 +364,10 @@ func (c *Controller) reconcileRequest(req *reconcileRequest) error {
 }
 
 func (c *Controller) readyForFailClose() bool {
+	//for remote clusters, we will set fail-close by default
```
Comment (howardjohn) [inline pkg/webhooks/validation/controller/controller.go:367]: It would be better to explain why we do this rather than what we are doing

**Review by howardjohn — COMMENTED** [review]
I think this makes sense, mentioning @stevenctl and @nmittler to give them a chance to find any issues.

File: pkg/webhooks/validation/controller/controller.go (hunk 2 above)
Comment (irisdingbj) [reply pkg/webhooks/validation/controller/controller.go]: fixed, PTAL, thanks.

**Comment by irisdingbj** [comment]
@stevenctl @nmittler this is used to support istiod-less as default remote as in https://github.com/istio/istio/pull/28181.  PTAL

**Review by howardjohn — APPROVED** [review]

**Pull request merged**